Fixed: cell properties dialog shows styles that are not allowed through allowedContent

engineering-this opened the pull request:
## What is the purpose of this pull request?

Bug fix

## Does your PR contain necessary tests?

All patches which change the editor code must include tests. You can always read more
on [PR testing](https://docs.ckeditor.com/ckeditor4/docs/#!/guide/dev_contributing_code-section-tests),
[how to set the testing environment](https://docs.ckeditor.com/ckeditor4/docs/#!/guide/dev_tests) and
[how to create tests](https://docs.ckeditor.com/ckeditor4/docs/#!/guide/dev_tests-section-creating-your-own-test)
in the official CKEditor documentation.

### This PR contains

- [x] Unit tests
- [x] Manual tests

## What changes did you make?

I've added `requiredContent` for each option in table cell properties dialog.

Closes #1986

engineering-this commented:
- I have rebased it on current master branch.
- Reworked how dialog is rendered. I think it looks much better now.
- Added manual test that allows to check various configurations.

engineering-this commented:
> - it's not possible to have dialog with only width or only height; it's possible only to have dialog with both of these settings;
As originally both width, and height inputs had `requiredContent: 'td{width,height}',`, but I guess you are right and it's ok to change it within this issue.

> - unit selection for height is broken.
➡️ #2084

Comandeer commented on tests/plugins/tabletools/manual/allowedcontent.md:
```diff
@@ -0,0 +1,22 @@
+@bender-tags: 4.10.1, 1986, bug
```
Please update version tag.

engineering-this commented:
Rebasted onto major, updated version tag and reworked unit test.

engineering-this commented:
Rebased and fixed failing test on IE8.

engineering-this commented:
- Changed dialog, so items will split into two columns evenly whenever there is at least 5 items, otherwise they will be in one column. 
- Dynamically removing spacers if they are last items.
- Removed spacers after bgColor and borderColor, fields will take all free space unless color dialog plugin is loaded. If it does dialog will look same as before.
- Updated tests after changes.
- Rebased.

engineering-this commented:
- Rebased onto latest major.
- Improved editor names in unit test, so each editor has both index and allowed rules included in name.
- Fixed color dialog in manual test.
- Added new spacer after alignment, which will only appear whenever last item from alignment group is not last in column.
- Fixed spacer after rowspan/colspan group, which appeared only if the second existed. Currently it appears whenever at least one of them is present.

Didn't mention it in comment before:

In previous push I've removed spacer between bgcolor and bordercolor, as they should be treated as one group.

Comandeer commented:
Rebased onto latest `master`.
